libres/lib/analysis/ies/ies.cpp

```diff
@@ -80,35 +80,33 @@ void ies::init_update(ies::data::Data &module_data,
     module_data.update_obs_mask(obs_mask);
 }
 
-void ies_initX__(const matrix_type *A, const matrix_type *Y0,
-                 const matrix_type *R, const matrix_type *E,
-                 const matrix_type *D, matrix_type *X,
+void ies_initX__(const std::optional<Eigen::MatrixXd> &A, const Eigen::MatrixXd &Y0,
+                 const Eigen::MatrixXd &R, const Eigen::MatrixXd &E,
+                 const Eigen::MatrixXd &D, Eigen::MatrixXd &X,
                  const ies::config::inversion_type ies_inversion,
                  const std::variant<double, int> &truncation,
-                 bool use_aa_projection, matrix_type *W0, double ies_steplength,
+                 bool use_aa_projection, Eigen::MatrixXd &W0, double ies_steplength,
                  int iteration_nr, double *costf)
 
 {
-    const int ens_size = matrix_get_columns(Y0);
-    const int nrobs_inp = matrix_get_rows(Y0);
+    const int ens_size = Y0.cols();
+    const int nrobs_inp = Y0.rows();
 
-    matrix_type *Y = matrix_alloc_copy(Y0);
-    matrix_type *H =
-        matrix_alloc(nrobs_inp, ens_size); // Innovation vector "H= S*W+D-Y"
-    matrix_type *S = matrix_alloc(
+    Eigen::MatrixXd Y = Y0;
+    Eigen::MatrixXd S = Eigen::MatrixXd::Zero(
         nrobs_inp,
         ens_size); // Predicted ensemble anomalies scaled with inv(Omeaga)
     const double nsc = 1.0 / sqrt(ens_size - 1.0);
```

Comment:
Nitpicking; suggesting to not define this variable and just do `1.0 / sqrt(ens_size - 1.0)` whenever needed.